edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Diagnostics/util/PrometheusMetricsParser.cs

```diff
@@ -13,51 +13,135 @@ namespace Microsoft.Azure.Devices.Edge.Agent.Diagnostics.Util
                     {
-                        double metricValue;
-                        if (!double.TryParse(match.Groups["metricvalue"]?.Value, out metricValue))
+                        continue;
+                    }
+
+                    string metricName = match.Groups["metricname"]?.Value ?? string.Empty;
+                    var tagNames = (match.Groups["labelname"].Captures as IEnumerable<Capture>).Select(c => c.Value);
+                    var tagValues =
+                        (match.Groups["labelvalue"].Captures as IEnumerable<Capture>)
+                        .Select(value =>
                         {
-                            continue;
-                        }
+                            // Evaluate escaped characters
+                            string result = string.Empty;
+                            bool escaped = false;
+                            foreach (char c in value.Value)
+                            {
+                                if (escaped)
+                                {
+                                    if (c == 'n')
+                                    {
+                                        result += "\n";
+                                    }
+                                    else
+                                    {
+                                        result += c;
+                                    }
+
+                                    escaped = false;
+                                }
+                                else if (c == '\\')
+                                {
+                                    escaped = true;
+                                }
+                                else
+                                {
+                                    result += c;
+                                }
+                            }
 
-                        string metricName = match.Groups["metricname"]?.Value ?? string.Empty;
-                        var tagNames = (match.Groups["tagname"].Captures as IEnumerable<Capture>).Select(c => c.Value);
-                        var tagValues = (match.Groups["tagvalue"].Captures as IEnumerable<Capture>).Select(c => c.Value);
+                            return result;
+                        });
 
-                        Dictionary<string, string> tags = tagNames.Zip(tagValues, (k, v) => new { k, v })
-                            .ToDictionary(x => x.k, x => x.v);
+                    Dictionary<string, string> tags = tagNames.Zip(tagValues, (k, v) => new { k, v })
+                        .ToDictionary(x => x.k, x => x.v);
+
+                    string metricTimestampString = match.Groups["metrictimestamp"]?.Value;
+                    if (metricTimestampString.Length != 0)
+                    {
+                        long metricTimestamp;
+                        if (!long.TryParse(metricTimestampString, out metricTimestamp))
+                        {
+                            continue;
```

Comment:
Same comment as above about silent continue

Reply:
Since the existing code already silently swallowed metrics that didn't match the regex at all, as well as metrics with bad metric values (as I explained in your previous comment), I figured it would make sense for this too.

Reply:
I've added warnings to all three cases (regex doesn't match, metric value is malformed, metric timestamp is malformed).